core/events.go

```diff
@@ -49,3 +52,318 @@ type ChainSideEvent struct {
 }
 
 type ChainHeadEvent struct{ Block *types.Block }
+
+type NewTxsFeed struct {
+	sync.RWMutex
+	subs []chan<- NewTxsEvent
+}
```

Comment:
I generally avoid embedding the mutex into a `struct` unless the `Lock()`/`Unlock()` methods are meant to be part of the exported API. Generally I like naming it `mu` unless other objects need to lock the `TxsFeed` externally.